futures-util/src/io/write_all.rs

```diff
@@ -33,10 +35,208 @@ impl<W: AsyncWrite + ?Sized + Unpin> Future for WriteAll<'_, W> {
                 this.buf = rest;
             }
             if n == 0 {
-                return Poll::Ready(Err(io::ErrorKind::WriteZero.into()))
+                return Poll::Ready(Err(io::ErrorKind::WriteZero.into()));
             }
         }
 
         Poll::Ready(Ok(()))
     }
 }
+
+/// Future for the
+/// [`write_all_vectored`](super::AsyncWriteExt::write_all_vectored) method.
+#[derive(Debug)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+#[cfg(feature = "write_all_vectored")]
```

Comment:
Could you split these into a new file(`futures-util/src/io/write_all_vectored.rs`) instead of using cfg for each item?

Reply:
Done.